dlnd/cime_config/stream_definition_dlnd.xml

```diff
@@ -19,7 +19,9 @@
       <meshfile>$LND_DOMAIN_MESH</meshfile>
     </stream_meshfile>
     <stream_datafiles>
-      <file first_year="0001" last_year="0030">$DLND_CPLHIST_DIR/$DLND_CPLHIST_CASE.cpl.hl2x1yr_glc.%y-01-01.nc</file>
+      <!-- Specify first year = 0, last year = 99999 in order to apply
+           this pattern to all possible years that might be included. -->
+      <file first_year="0" last_year="99999">$DLND_CPLHIST_DIR/$DLND_CPLHIST_CASE.cpl.hl2x1yr_glc.%y-01-01.nc</file>
```

Comment:
@mvertens it threw me off that I needed to change this in addition to changing `DLND_CPLHIST_YR_START` and `DLND_CPLHIST_YR_END`. I didn't like that these two things needed to be kept in sync: it felt error-prone and less flexible to possible user modifications. The easiest way I could see around it was to generalize the year match here.